test/realtime/vehicle_test.exs

```diff
@@ -579,14 +581,21 @@ defmodule Realtime.VehicleTest do
       assert Vehicle.end_of_trip_type(block, last_trip_of_block, "run2", "middle") == :pull_back
     end
 
+    test "when finished with the last trip of the block but still logged in, returns :pull_back",
```

Comment:
question: This test name specifies "still logged in", but the change to `end_of_trip_type` doesn't seem to consider whether or not a vehicle is logged in. Should it?

Reply:
It sort of implicitly does because `end_of_trip_type/4` defaults to `:another_trip` when `run_id` is null, so the fact that this test sets a non-null run ID means that the vehicle must be logged in.